### PR TITLE
Automatically create draft release

### PR DESCRIPTION
**Pull request by abelfodil**
The first part of this PR:
- Automatically creates a draft release whenever there's a push on master 
- Fills it with the pending (between previous and new release) PRs 
- Adds mobile and server builds

Here's a [published](https://github.com/PolyCortex/polydodo/releases/tag/v0.0.1) release and a [draft](https://github.com/PolyCortex/polydodo/releases/tag/untagged-b10887a263ed68bad598) (unpublished) release. Note that no element in the draft release has been added manually (including the title).

The second part of this PR:
- Fetches the latest release
- Displays a list of downloadable assets on the server download page

The download page looks like this :

image

### CODE REVIEW

File: web/src/views/analyze_sleep/waiting_for_server/index.js [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+  return (
+    <Container>
+      <h3 className="mb-4">Waiting for local server to be running...</h3>
+      <Row>
+        <Col className="text-center mb-4">
+          <Spinner style={{ width: '3rem', height: '3rem' }} />
+        </Col>
+      </Row>
+
+      <h3>Download latest server releases</h3>
```
Comment (abelfodil) [inline web/src/views/analyze_sleep/waiting_for_server/index.js:20]:
```suggestion
      <h3>Download latest server release</h3>
```

File: .github/workflows/backend.yml [hunk 2]
```diff
@@ -57,3 +50,54 @@ jobs:
...
+        with:
+          name: server_Linux
+          path: backend/dist/linux
+      - uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./backend/dist/linux/app
+          asset_name: server-linux
```
Comment (WilliamHarvey97) [inline .github/workflows/backend.yml:76]: Peut-être mettre plutôt un nom tel que `polydodo_server_standalone_linux-x86` ?

File: .github/workflows/backend.yml [hunk 3]
```diff
@@ -57,3 +50,54 @@ jobs:
...
+        with:
+          name: server_Windows
+          path: backend/dist/windows
+      - uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./backend/dist/windows/app.exe
+          asset_name: server-windows.exe
```
Comment (WilliamHarvey97) [inline .github/workflows/backend.yml:89]: Same here

File: .github/workflows/backend.yml [hunk 4]
```diff
@@ -57,3 +50,54 @@ jobs:
...
+        with:
+          name: server_macOS
+          path: backend/dist/macos
+      - uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./backend/dist/macos/app
+          asset_name: server-macos
```
Comment (WilliamHarvey97) [inline .github/workflows/backend.yml:102]: and here

File: .github/workflows/mobile.yml [hunk 5]
```diff
@@ -29,11 +29,32 @@ jobs:
...
+          path: mobile/out
+      - uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          name: iOS app
-          path: mobile/build/ios/iphoneos/Runner.app
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./mobile/out/app-release.apk
+          asset_name: app-android.apk
```
Comment (WilliamHarvey97) [inline .github/workflows/mobile.yml:59]: Same for that file

File: web/src/requests/latest_release.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+const octokit = new Octokit();
+
+const getPublishedReleases = async () => {
+  const releases = await octokit.repos.listReleases({ owner: 'polycortex', repo: 'polydodo' });
+  const publishedReleases = releases.data.filter((release) => !release.draft);
+  return publishedReleases;
+};
+
+const assetDisplayNameMap = {
```
Comment (WilliamHarvey97) [inline web/src/requests/latest_release.js:12]: Place global constants at the beginning of file.

**Review by WilliamHarvey97 — APPROVED** [review]
clean!

Maybe we should add a wiki page to document all this good work! :+1:

**Pull request merged**